### PR TITLE
engine token precision

### LINKED ISSUE #2623: Bug: Engine token precision need fixing
https://discord.com/channels/@me/920267778190086205/1072045165579083787

### PR DESCRIPTION
**Pull request by aliseyalvi**
### What does this PR?
This PR add precision in engine tokens

### Issue number
fixes #2623 

### Screenshots/Video
image

image
image

### CODE REVIEW

File: src/screens/assetDetails/children/coinSummary.tsx [hunk 1]
```diff
@@ -22,12 +22,12 @@ export const CoinSummary = ({
...
   onInfoPress,
 }: CoinSummaryProps) => {
-  const { balance, estimateValue, savings, extraDataPairs, actions } = coinData;
+  const { balance, estimateValue, savings, extraDataPairs, actions, precision } = coinData;
 
   const valuePairs = [
     {
       dataKey: 'amount_desc',
-      value: balance,
+      value: balance.toFixed(precision),
```
Comment (noumantahir) [inline src/screens/assetDetails/children/coinSummary.tsx:30]: @aliseyalvi are we setting any default value for precision, how are we managing non engine precision here.?

**Review by noumantahir — CHANGES_REQUESTED** [review]

File: src/screens/assetDetails/children/coinSummary.tsx (hunk 1 above)
Comment (aliseyalvi) [reply src/screens/assetDetails/children/coinSummary.tsx:30]: @noumantahir the above change only affects engine tokens and we are considering all engine tokens returns precision. We can add default precision here as well to handle null case

Comment (aliseyalvi) [reply src/screens/assetDetails/children/coinSummary.tsx]: @noumantahir added default precision to 3 in coinSummary
https://github.com/ecency/ecency-mobile/pull/2624/commits/f82eedd9d0add26effa6ba3154996175585a75d3

**Review by noumantahir — APPROVED** [review]
lgtm

**Review by feruzm — APPROVED** [review]

**Pull request merged**